falcon/request.py

```diff
@@ -683,6 +687,22 @@ def access_route(self):
     def remote_addr(self):
         return self.env.get('REMOTE_ADDR')
 
+    @property
+    def port(self):
+        try:
+            host_header = self.env['HTTP_HOST']
+            host, port = parse_host(host_header)
+        except KeyError:
+            port = self.env.get('SERVER_PORT')
+        return port
+
+    @property
+    def netloc(self):
+        try:
+            return self.env['HTTP_HOST']
+        except KeyError:
+            return self.host + ':' + self.port
```

Comment:
Since we already know the `HTTP_HOST` key does not exist, the redundant checks in self.host and self.port seem redundant. As an alternative, what do you think about factoring out [this section](https://github.com/falconry/falcon/blob/master/falcon/request.py#L554-L569) from the `uri` property to use as the implementation of `netloc`, then use `netloc` from `uri()` as per below?

```py
@property
def uri(self):
    if self._cached_uri is None:
        env = self.env
        protocol = env['wsgi.url_scheme']

        # PERF: For small numbers of items, '+' is faster
        # than ''.join(...). Concatenation is also generally
        # faster than formatting.
        value = (protocol + '://' +
                 self.netloc +
                 self.app +
                 self.path)

        if self.query_string:
            value = value + '?' + self.query_string

        self._cached_uri = value

    return self._cached_uri
```